### ISSUE TITLE
Assign different arguments to different targets in the batch mode

### ISSUE THREAD

**Issue by NiBoyang** [ccsb-scripps/AutoDock-GPU#206]
Hi there. Now I'm using the batch mode to virtually screen a super large database. I wonder if there is any way that I can assign different arguments to some targets in the batch? For instance, I'm currently using 'nrun=100' and the batch mode, so for each target in the batch it will keep running using 'nrun=100', but for some small compounds in the batch I don't want such many runs, and how can I set it up for them separately.

**Comment by diogomart** [comment]
Hi, that isn't possible, but AutoDock-GPU stops when it detects convergence, so setting up the search effort for each molecule manually is unlikely to improve the performance/runtime ratio. Molecules with many rotatable bonds (> 20) are likely to fail even if the  search effort is extended beyond the defaults.

**Comment by NiBoyang** [comment]
Hi, thanks for the reply. I wonder what the "convergence" exactly is, is that the standard deviation of the energy? Also I just checked the "timing" file returned by the batch mode. I wonder why the "n_evals" changes according to the number of rotatable bonds as I didn't change the default setting of "--nev" and it should be the 2500000 per LGA run, but for some compounds with more than 20 rotatable bonds I saw their "n_evals" were going beyond that number. Moreover, the default setting for AutoStop should be "on", but why some "nev_at_stop" surpassed the "n_evals" and the "autostopped" returned 0.

**Comment by diogomart** [comment]
Convergence, in this context, is when we *probably* found the pose with the best (lowest) possible score. The population of the genetic algorithm will have many poses with bad (high) scores but we don't care about those. With `--heuristics 1`, which is the default, the maximum number of evals does indeed increase with the number of rotatable bonds. The docking stops when either convergence is detected or when the maximum number of evals is reached. If you really want to set the number of evals manually with `--nev`, then set also `-A 0 -H 0`.

**Comment by NiBoyang** [comment]
Hi, thanks for the reply. I wonder if it's possible to write the parallelization using Python's multiprocessing to achieve the same effect as the batch mode?

**Comment by diogomart** [comment]
No, we would need Python bindings to relatively low-level functions for that to be possible.
